File: JavaWorld/src/com/java/leetcode/LongestCommonPrefix.java
package com.java.leetcode;

public class LongestCommonPrefix {

    public static void main(String... args) {
        String[] input = {"Flight", "Flower", "Fly", "Fortunate", "Flood"};
        System.out.println(longestCommonPrefix(input));
    }

    public static String longestCommonPrefix(String[] input) {
        if(input == null || input.length < 1) return "";
        if(input.length == 1) return input[0];

        String ret = input[0];
        int idx = 1;
        while(!ret.isEmpty() && idx < input.length) {
            if(input[idx].indexOf(ret) != 0) {
                ret = ret.substring(0, ret.length()-1);
            } else {
                idx++;
            }
        }

        return ret;
    }
}

File: JavaWorld/src/com/kotlin/LongestCommonPrefix.kt
package com.kotlin

fun main() {
    val input = arrayOf("Flight", "Flower", "Flood")
    println(longestCommonPrefix(input))
}

fun longestCommonPrefix(input: Array<String>): String {
    if(input.isEmpty()) return ""
    if(input.size == 1) return input[0]

    var prefix = input[0]
    for(i in 1 until input.size) {
        while(input[i].indexOf(prefix) != 0) {
            prefix = prefix.substring(0, prefix.length-1)
            if(prefix.isEmpty()) return ""
        }
    }

    return prefix
}


